src/tree/newtree/provisional_tree.rs

```diff
@@ -0,0 +1,162 @@
+use std::collections::HashMap;
+
+use hpke::HPKEPrivateKey;
+
+use crate::messages::proposals::ProposalID;
+use crate::messages::proposals::ProposalQueue;
+use crate::prelude::Ciphersuite;
+use crate::prelude::KeyPackageBundle;
+use crate::schedule::EpochSecrets;
+use crate::tree::index::NodeIndex;
+use crate::tree::InvitationList;
+use crate::tree::TreeError;
+
+use super::binary_tree::BinaryTree;
+use super::node::Node;
+use super::node::NodeContents;
+use super::tree_secrets::CommitSecret;
+use super::tree_secrets::PathSecret;
+use super::RatchetTreeBundle;
+
+#[allow(dead_code)]
+pub(crate) struct ProvisionalTree<'a> {
+    /// A reference to the RatchetTreeBundle from which the `ProvisionalTree`
+    /// was created.
+    original_tree: Option<&'a RatchetTreeBundle>,
+
+    /// If the `original_tree` is `Some`, the `node_index` should be equal to
+    /// `original_tree.node_index`. If the `original_tree` is `None`, it has to
+    /// be provided when creating the provisional tree.
+    own_node_index: NodeIndex,
```

Comment:
This should also come from the original tree, since members cannot move around in the tree.